sql/hive/src/main/scala/org/apache/spark/sql/hive/client/IsolatedClientLoader.scala

```diff
@@ -124,6 +124,7 @@ private[hive] class IsolatedClientLoader(
     name.contains("slf4j") ||
     name.contains("log4j") ||
     name.startsWith("org.apache.spark.") ||
+    (name.startsWith("org.apache.hadoop.") && !name.startsWith("org.apache.hadoop.hive.")) ||
```

Comment:
scope is too huge, and without reload hadoop, will only have one FileSystem.Cache which will cause [SPARK-11083](https://issues.apache.org/jira/browse/SPARK-11083)